lib/src/arithmetic/fixed_point/fixed_point_logic.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_logic.dart
+// Implementation of Fixed Point objects
+//
+// 2024 September 25
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/exceptions.dart';
+
+/// Fixed point logic representation
+class FixedPoint extends Logic {
+  /// [integerWidth] is the width of bits reserved for integer part
+  final int integerWidth;
+
+  /// [fractionWidth] is the width of bits reserved for fractional part
+  final int fractionWidth;
+
+  static int _fixedPointWidth(int a, int b) => 1 + a + b;
+
+  /// [FixedPoint] Constructor
+  FixedPoint(
+      {required this.integerWidth,
+      required this.fractionWidth,
+      super.name,
+      super.naming})
+      : super(width: _fixedPointWidth(integerWidth, fractionWidth)) {
+    if (integerWidth < 0) {
+      throw RohdHclException('integerWidth must be non-negative');
+    }
+    if (fractionWidth < 0) {
+      throw RohdHclException('fractionWidth must be non-negative');
+    }
+  }
+
+  /// Clone for I/O
+  FixedPoint clone({String? name}) => FixedPoint(
+        integerWidth: integerWidth,
+        fractionWidth: fractionWidth,
+      );
```

Comment:
should we add some operator overloads in here so that `+` etc. returns the right fixed-point stuff?